### ISSUE TITLE
Bug report- public_ip with zones attribute NULL ignored

### ISSUE THREAD

**Issue by sjackson0109** [aztfmod/terraform-azurerm-caf#1748]
### Versions:
terraform -v
Terraform v1.4.6
on windows_amd64
+ provider registry.terraform.io/aztfmod/azurecaf v1.2.26
+ provider registry.terraform.io/azure/azapi v1.6.0
+ provider registry.terraform.io/hashicorp/azuread v2.39.0
+ provider registry.terraform.io/hashicorp/azurerm v3.56.0
+ provider registry.terraform.io/hashicorp/external v2.3.1
+ provider registry.terraform.io/hashicorp/local v2.4.0
+ provider registry.terraform.io/hashicorp/null v3.2.1
+ provider registry.terraform.io/hashicorp/random v3.3.2
+ provider registry.terraform.io/hashicorp/time v0.9.1
+ provider registry.terraform.io/hashicorp/tls v4.0.4

### Affected Resource(s)/Data Source(s)
azurerm_public_ip

### Terraform Configuration Files
```hcl
public_ip_addresses = {
  ukw_natgw_pip = {
    name                    = "ukw-hub-natgw"
    resource_group_key      = "hub"
    region                  = "ukw"    ## NOTE UKWEST REGION DOES NOT SUPPORT ZONE BASED DEPLOYMENTS
    sku                     = "Standard"
    allocation_method       = "Static" 
    ip_version              = "IPv4"
    idle_timeout_in_minutes = "4"
    zones                   = null     ## FORCE NO ZONES TO BE DECLARED
  }
}
```

### Expected Behaviour

`terraform plan` should set `zones = null`. Clearly it does not!

Note: According to the aztfmod sub-module, located (here)[https://github.com/aztfmod/terraform-azurerm-caf/blob/main/modules/networking/public_ip_addresses/variables.tf], the declared variable `zones` looks like this:
```
variable "zones" {
  description = "The availability zone to allocate the Public IP in. Possible values are 1, 2, 3. Defaults to null."
  type        = list(string)
  nullable    = true

  validation {
    condition     = setunion(["1", "2", "3"], var.zones == null ? [""] : var.zones) == toset(["1", "2", "3"])
    error_message = "Update behavior to mandate specifying the zones in order to avoid accidental resources destructions. https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/public_ip#zone."
  }
}
```

### Actual Behaviour
```
Terraform will perform the following actions:

  # module.aztfmod.module.public_ip_addresses["ukw_hub_natgw_pip"].azurerm_public_ip.pip will be created
  + resource "azurerm_public_ip" "pip" {
      + allocation_method       = "Static"
      + ddos_protection_mode    = "VirtualNetworkInherited"
      + fqdn                    = (known after apply)
      + id                      = (known after apply)
      + idle_timeout_in_minutes = 4
      + ip_address              = (known after apply)
      + ip_version              = "IPv4"
      + location                = "ukwest"
      + name                    = "ukw-hub-natgw"
      + resource_group_name     = "HUB"
      + sku                     = "Standard"
      + sku_tier                = "Regional"
      + zones                   = [
          + "1",
          + "2",
          + "3",
        ]
    }

Plan: 1 to add, 0 to change, 0 to destroy.
```

>&#10071; take note of the `zones` attribute above!

Terraform apply returns:

```
│ Error: creating/updating Public Ip Address: (Name "ukw-hub-natgw" / Resource Group "GSUKHUB"): network.PublicIPAddressesClient#CreateOrUpdate: Failure sending request: StatusCode=400 -- Original Error: Code="LocationNotSupportAvailabilityZones" Message="The resource 'Microsoft.Network/publicIPAddresses/ukw-hub-natgw' does not support availability zones at location 'ukwest'."
│
│   with module.aztfmod.module.public_ip_addresses["ukw_natgw_pip"].azurerm_public_ip.pip,
│   on .terraform\modules\aztfmod\modules\networking\public_ip_addresses\module.tf line 3, in resource "azurerm_public_ip" "pip":
│    3: resource "azurerm_public_ip" "pip" {
```

### Steps to Reproduce
```
terraform plan
terraform apply -auto-approve
```

**Comment by sjackson0109** [comment]
The aztfmod submodule's variable description includes the words `Defaults to null.` yet there is no `default = null` property inside the variable.

Setting the value to null yields:
```
public _ip_addresses = {
ukw_natgw_pip = {
    name                     = "ukw-hub-natgw"
    resource_group_key      = "hub"
    region                  = "ukw"
    sku                     = "Standard"
    allocation_method       = "Static" 
    ip_version              = "IPv4"
    idle_timeout_in_minutes = "4"
    zones                   = null
  }
```
terraform plan:
```
 Error: Invalid value for variable
│
│   on .terraform\modules\aztfmod\networking.tf line 143, in module "public_ip_addresses":
│  143:   zones = try(each.value.sku, "Basic") == "Basic" ? [] : try(each.value.zones, null) == null ? ["1", "2", "3"] : each.value.zones
│     ├────────────────
│     │ var.zones is list of string with 3 elements
│
│ Update behavior to mandate specifying the zones in order to avoid accidental resources destructions.
│ https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/public_ip#zone.
│
│ This was checked by the validation rule at .terraform\modules\aztfmod\modules\networking\public_ip_addresses\variables.tf:88,3-13.
```


Assessling line 143...
```
zones = try(each.value.sku, "Basic") == "Basic" ? [] : try(each.value.zones, null) == null ? ["1", "2", "3"] : each.value.zones
```

My sku="Standard", so the first match is invalid.. leaving this code:
```
try(each.value.zones, null) == null ? ["1", "2", "3"] : each.value.zones
```
My zones=null, so the try function returns NULL.
the comparison is true returning `["1", "2", "3"]`, matching what we see in the plan (invalid)

As described in azurerm_public_ip:
"Availability Zones are only supported with a [Standard SKU](https://docs.microsoft.com/azure/virtual-network/virtual-network-ip-addresses-overview-arm#standard) and [in select regions](https://docs.microsoft.com/azure/availability-zones/az-overview) at this time. Standard SKU Public IP Addresses that do not specify a zone are not zone-redundant by default."

This module should be setting NULL as the default value, not replacing null with the set of values!

**Comment by sjackson0109** [comment]
I've fixed this in my offline copy - but cannot push to a new branch in this repo (access denied).

Replace line #143 of `.\modules\aztfmod\networking.tf` with:
`  zones = try(each.value.sku, "Basic") == "Basic" ? [] : try(each.value.zones, null) == null ? ["1", "2", "3"] : each.value.zones`

and replace line #89 of `.\modules\aztfmod\modules\networking\public_ip_addresses\variables.tf` with:
`    condition     = setunion(["1", "2", "3"], var.zones == null ? [""] : var.zones) == toset(["1", "2", "3"])`

If someone is willing to grant me permission to branch/commit somewhere, i'd be happy to do it.

**Comment by sjackson0109** [comment]
Definitely resolved in my offline version, i published and released online for testing on v5.7.3, but forgot to include one of the file-updates.  Published again as v5.7.4 and this now creates UKSouth public IPs with `zones=null`.

Revised the logic further. If we wanted `zones=null` to be set for a public ip deployed in UKSouth, then the module was replacing the zone with `["1","2","3"]`. This shouldn't be the case. 
Published again as v5.7.6 - and tested half a dozen combinations of zones, all working in multiple regions (zonal supported and not).
